src/ahbicht/content_evaluation/content_evaluation_result.py

```diff
@@ -107,6 +108,52 @@ def sanitize(self) -> None:
         self._remove_duplicates()
         self._sort_keys()
 
+    def generate_possible_content_evaluation_results(self) -> List[ContentEvaluationResult]:
+        """
+        A categorized key extract allows to generate nearly all possible content evaluation results,
+        except for hints, error messages, resolving packages.
+        """
+        results: List[ContentEvaluationResult] = []
+        if len(self.format_constraint_keys) == 0 and len(self.requirement_constraint_keys) == 0:
+            return results
+        # for easier debugging below, replace the generators "(" with a materialized lists "["
+        if len(self.format_constraint_keys) > 0:
+            possible_fcs = (
+                z
+                for z in combinations(
+                    product(self.format_constraint_keys, [True, False]), len(self.format_constraint_keys)
+                )
+                if len({y[0] for y in z}) == len(self.format_constraint_keys)
```

Comment:
Kannst du mir die Zeile noch erklären? In welchem Fall wäre die Bedingung nicht erfüllt?

Reply:
[ohoh](https://pep20.org/#hard)

Reply:
fe4f9bbd0a87c52cfb8d1a04ffd01d2bb9b5653d <-- das sollte passen denke ich.